server/bin/basket-proxy-server.js

```diff
@@ -0,0 +1,180 @@
+
+var config = require('../lib/configuration');
+mozlog.config(config.get('logging'));
+
+var logger = require('mozlog')('server.basketproxy');
+
+var express = require('express');
+var bodyParser = require('body-parser');
+var cors = require('cors');
+
+var CORS_ORIGIN = config.get('public_url');
+var API_KEY = config.get('basket.api_key');
+var API_URL = config.get('basket.api_url');
+var VERIFY_URL = config.get('oauth_url') + '/v1/verify';
+
+
+// Error codes are defined in:
+// https://github.com/mozilla/basket-client/blob/master/basket/errors.py
+var BASKET_ERRORS = {
+  NETWORK_FAILURE: 1,
+  INVALID_EMAIL: 2,
+  UNKNOWN_EMAIL: 3,
+  UNKNOWN_TOKEN: 4,
+  USAGE_ERROR: 5,
+  EMAIL_PROVIDER_AUTH_FAILURE: 6,
+  AUTH_ERROR: 7,
+  SSL_REQUIRED: 8,
+  INVALID_NEWSLETTER: 9,
+  INVALID_LANGUAGE: 10,
+  EMAIL_NOT_CHANGED: 11,
+  CHANGE_REQUEST_NOT_FOUND: 12,
+
+  // If you get this, report it as a bug so we can add a more specific
+  // error code.
+  UNKNOWN_ERROR: 99
+};
+
+function errorResponse(desc, code) {
+  // Format from
+  // https://basket.readthedocs.org/en/latest/newsletter_api.html
+  return {
+    status: 'error',
+    desc: String(desc),
+    code: code || BASKET_ERRORS.UNKNOWN_ERROR
+  };
+}
+
+// Verify an OAuth token and return the associated credentials
+function verifyOAuthToken() {
+  return function (req, res, next) {
+    var authHeader = req.headers && req.headers.authorization;
+
+    if (! authHeader) {
+      logger.error('auth.missing-authorization-header');
+      res.status(400).json(errorResponse('missing authorization header', BASKET_ERRORS.USAGE_ERROR));
+      return;
+    }
+
+    var token = authHeader.replace(/^Bearer /, '');
+
+    request.post({
+      url: VERIFY_URL,
+      json: {
+        token: token
+      }
+    }, function (err, _, body) {
+      // TODO use an actual error format
+      if (err) {
+        logger.error('auth.error', err);
+        res.status(400).json(errorResponse(err, BASKET_ERRORS.UNKNOWN_ERROR));
+        return;
+      }
+
+      if (body.code >= 400) {
+        logger.error('auth.unauthorized', body);
+        res.status(body.code).json(errorResponse('unauthorized', BASKET_ERRORS.AUTH_ERROR));
+        return;
+      }
+
+      logger.debug('auth.valid', body);
+
+      res.locals.creds = body;
+
+      next();
+    });
+  };
+}
+
+// Send a request to the Basket backend
+function basketRequest(path, method, params, done) {
+  var req = request({
+    url: API_URL + path,
+    strictSSL: true,
+    method: method,
+    headers: {
+      'X-API-Key': API_KEY
+    },
+    form: params
+  }, done);
+
+  return req;
+}
+
+function initApp() {
+  var app = express();
+  app.use(bodyParser.json());
+
+  app.use(cors({
+    origin: CORS_ORIGIN
+  }));
+
+  app.use(verifyOAuthToken());
+
+  app.get('/lookup-user', function (req, res) {
+    var params = req.body;
+    var creds = res.locals.creds;
+
+    basketRequest('/lookup-user/?email=' + creds.email, 'get', params).pipe(res);
```

Comment:
Yeah, so this needs to change here and other places where we just pipe through the response. It needs to log errors at minimum, but really should be standard HTTP response code logging for success and failure. 

d